lib/src/crypto_subtle.dart

```diff
@@ -116,148 +128,387 @@ class CryptoKeyPair {
+  TypedData get label => getProperty(_jsObj, 'label');
+  String get namedCurve => getProperty(_jsObj, 'namedCurve');
+  CryptoKey get public => CryptoKey(getProperty(_jsObj, 'public'));
+  TypedData get counter => getProperty(_jsObj, 'counter');
+  int get length => getProperty(_jsObj, 'length');
+  TypedData get iv => getProperty(_jsObj, 'iv');
+  TypedData get additionalData => getProperty(_jsObj, 'additionalData');
+  int get tagLength => getProperty(_jsObj, 'tagLength');
+  TypedData get salt => getProperty(_jsObj, 'salt');
+  TypedData get info => getProperty(_jsObj, 'info');
+  int get iterations => getProperty(_jsObj, 'iterations');
+
+  Algorithm({
+    String? name,
+    int? modulusLength,
+    Uint8List? publicExponent,
+    String? hash,
+    int? saltLength,
+    TypedData? label,
+    String? namedCurve,
+    CryptoKey? public,
+    TypedData? counter,
+    int? length,
+    TypedData? iv,
+    TypedData? additionalData,
+    int? tagLength,
+    TypedData? salt,
+    TypedData? info,
+    int? iterations,
+  }) : _jsObj = newObject() {
+    if (name != null) {
+      setProperty(_jsObj, 'name', name);
+    }
+    if (modulusLength != null) {
+      setProperty(_jsObj, 'modulusLength', modulusLength);
+    }
+    if (publicExponent != null) {
+      setProperty(_jsObj, 'publicExponent', publicExponent);
+    }
+    if (hash != null) {
+      setProperty(_jsObj, 'hash', hash);
+    }
+    if (saltLength != null) {
+      setProperty(_jsObj, 'saltLength', saltLength);
+    }
+    if (label != null) {
+      setProperty(_jsObj, 'label', label);
+    }
+    if (namedCurve != null) {
+      setProperty(_jsObj, 'namedCurve', namedCurve);
+    }
+    if (public != null) {
+      setProperty(_jsObj, 'public', public._jsObj);
+    }
+    if (counter != null) {
+      setProperty(_jsObj, 'counter', counter);
+    }
+    if (length != null) {
+      setProperty(_jsObj, 'length', length);
+    }
+    if (iv != null) {
+      setProperty(_jsObj, 'iv', iv);
+    }
+    if (additionalData != null) {
+      setProperty(_jsObj, 'additionalData', additionalData);
+    }
+    if (tagLength != null) {
+      setProperty(_jsObj, 'tagLength', tagLength);
+    }
+    if (salt != null) {
+      setProperty(_jsObj, 'salt', salt);
+    }
+    if (info != null) {
+      setProperty(_jsObj, 'info', info);
+    }
+    if (iterations != null) {
+      setProperty(_jsObj, 'iterations', iterations);
+    }
+  }
 }
 
-JsonWebKey jsonWebKeyFromJs(dynamic obj) {
-  final json = jsonDecode(_stringify(obj));
-  try {
-    return JsonWebKey.fromJson(json);
-  } on FormatException catch (e) {
-    throw UnsupportedError(
-      'exported JsonWebKey is not valid, this could be internal error or '
-      'lacking browser support -- problem: ${e.message}',
-    );
+/// Create [JsonWebKey] from [jsObj].
+JsonWebKey _jsonWebKeyFromJsObj(Object jsObj) {
+  final jwk = JsonWebKey();
+
+  if (hasProperty(jsObj, 'kty')) {
```

Comment:
Perhaps make a datastructure for all these. It seems too easy for a mistake to slip through.

```
void setPropIfPresent(String name, void Function(Object?) setter) { 
  if (hasProperty(jsObj, name)) {
    setter(getProperty(jsObj, name);
  }
}

setPropIfPresent('kty', (e) => jwk.kty = e);
setPropIfPresent('use', (e) => jwk.use = e);
...

Reply:
A utility like `setPropIfPresent` would certainly make sense